### PR TITLE
TRUNK-3253: Add junit test for searching on voided patients

### PR DESCRIPTION
**Pull request by gitcliff**
link:https://issues.openmrs.org/browse/TRUNK-3253
Added junit test for searching on voided patients

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/db/PatientDAOTest.java [hunk 1]
```diff
@@ -755,6 +755,42 @@ public void getPatients_shouldNotMatchVoidedPatientNames_SignatureNo1() {
 		Assert.assertEquals(0, patients.size());
 	}
 	
+	@Test
+	public void getPatients_shouldNotMatchVoidedPatientsWhenVoidedFalseIsPassed() {
+		List<Patient> patients = dao.getPatients("Oloo", 0, 11);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/PatientDAOTest.java]: What does 0 in the second parameter mean?

File: api/src/test/java/org/openmrs/api/db/PatientDAOTest.java [hunk 2]
```diff
@@ -755,6 +755,42 @@ public void getPatients_shouldNotMatchVoidedPatientNames_SignatureNo1() {
...
+		}
+		dao.savePatient(patient);
+		updateSearchIndex();
+		patients = dao.getPatients("Oloo", false, 0, 11);
+		Assert.assertEquals(0, patients.size());
+	}
+	
+	@Test
+	public void getPatients_shouldMatchVoidedPersonWhenVoidedTrueIsPassed() {
+		List<Patient> patients = dao.getPatients("Oloo", 0, 11);
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/db/PatientDAOTest.java]: What does 0 in the second parameter mean?

Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/PatientDAOTest.java]: @dkayiwa its an integer parameter representing the starting position and in this case the specified number is 0

File: api/src/test/java/org/openmrs/api/db/PatientDAOTest.java (hunk 1 above)
Comment (gitcliff) [reply api/src/test/java/org/openmrs/api/db/PatientDAOTest.java]: @dkayiwa its an integer parameter representing the starting position and in this case the specified number is 0

Comment (ibacher) [reply api/src/test/java/org/openmrs/api/db/PatientDAOTest.java:760]: @gitcliff Are you still interested in working on this? If so, this should be updated to:

```suggestion
		List<Patient> patients = dao.getPatients("Oloo", false, 0, 11);
```

**Review by ibacher — APPROVED** [review]

**Pull request merged**